### PR TITLE
[Addressing] Documentation update of Addressing component

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q             | A
| ------------- | ---
| Doc fix?      | yes
| New docs?     | no
| Fixed tickets | -

### CODE REVIEW

File: components/Addressing/index.rst [hunk 1]
```diff
@@ -1,14 +1,14 @@
 Addressing
 ==========
 
-Sylius Addressing component for PHP E-Commerce applications.
-
-The component provides you with basic Address, Country, Province and Zone models.
+Sylius Addressing component for PHP E-Commerce applications which
+provides You with basic Address, Country, Province and Zone models.
```
Comment (TheMadeleine) [inline components/Addressing/index.rst]: we don't need to use capital first letter for `you` :)

File: components/Addressing/interfaces.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+   For more detailed information go to `Sylius API AddressInterface`_.
+
+.. _Sylius API AddressInterface: http://api.sylius.org/Sylius/Component/Addressing/Model/AddressInterface.html
+
+.. _component_addressing_model_country-interface:
+
+CountryInterface
+~~~~~~~~~~~~~~~~
+
+This interfaces should be implemented by a model representing a country.
```
Comment (TheMadeleine) [inline components/Addressing/interfaces.rst]: `This interface` - singular. and `by models representing a **Country**`

File: components/Addressing/interfaces.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+.. _Sylius API ZoneInterface: http://api.sylius.org/Sylius/Component/Addressing/Model/ZoneInterface.html
+
+
+.. _component_addressing_model_zone-member-interface:
+
+ZoneMemberInterface
+~~~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by any model you would like to characterize, as something a specific
```
Comment (TheMadeleine) [inline components/Addressing/interfaces.rst]: `This interface should be implemented by models that represent something that can be a memeber of a **Zone**, e.g. all countries in the European Union.`

File: components/Addressing/interfaces.rst (hunk 2 above)
Comment (NeverResponse) [inline components/Addressing/interfaces.rst]: Agree with plural for models, but have to disagree with the bolding as i only use bold for referring to classes already implemented by default or as an emphasis to important words in basic usage.

File: components/Addressing/models.rst [hunk 4]
```diff
@@ -1,262 +1,222 @@
...
++-----------+--------------------------------------+
+| isoName   | Country's ISO code                   |
++-----------+--------------------------------------+
+| provinces | Collection of **Province** objects   |
++-----------+--------------------------------------+
+| enabled   | Indicates whether country is enabled |
++-----------+--------------------------------------+
+| createdAt | Date when the object was created     |
++-----------+--------------------------------------+
+| updatedAt | Date of last object's update         |
```
Comment (TheMadeleine) [inline components/Addressing/models.rst]: Actually.. it does not implement `TimestampableInterface`, so it does not have createdAt and updatedAt fields.

File: components/Addressing/models.rst [hunk 5]
```diff
@@ -1,262 +1,222 @@
...
 
-    <?php
+ZoneMemberProvince
+------------------
 
-    use Sylius\Component\Addressing\Country;
-    use Sylius\Component\Addressing\Zone;
-    use Sylius\Component\Addressing\ZoneInterface;
-    use Sylius\Component\Addressing\ZoneMemberCountry;
+Province member of a zone is represented as a **ZoneMemberProvince** instance.
```
Comment (TheMadeleine) [inline components/Addressing/models.rst]: represented by *

Comment (NeverResponse) [inline components/Addressing/models.rst]: Just checked, everywhere in **ZoneMember** area, everything above is alright. My best guess is, there was another word which i changed and forgot about it ._.

**Comment by pjedrzejewski** [comment]
Nice, looks good! :+1:

File: components/Addressing/interfaces.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+ZoneMatcherInterface
+~~~~~~~~~~~~~~~~~~~~
+
+This interface should be implemented by a service responsible of finding the best matching zone,
+and all zones containing the provided :ref:`component_addressing_model_address`.
+
+.. note::
+   For more detailed information go to `Sylius API ZoneMatcherInterface`_.
+
+.. _Sylius API ZoneMatcherInterface: http://api.sylius.org/Sylius/Component/Addressing/Model/ZoneMatcherInterface.html
```
Comment (michalmarcinkowski) [inline components/Addressing/interfaces.rst]: ...api.sylius.org/Sylius/Component/Addressing/**Matcher**/ZoneMatcherInterface...

File: components/Addressing/interfaces.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+RestrictedZoneCheckerInterface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A service implementing this interface should be able to check
+if given :ref:`component_addressing_model_address` is in a restricted zone.
+
+.. note::
+   For more detailed information go to `Sylius API RestrictedZoneCheckerInterface`_.
+
+.. _Sylius API RestrictedZoneCheckerInterface: http://api.sylius.org/Sylius/Component/Addressing/Model/RestrictedZoneCheckerInterface.html
```
Comment (michalmarcinkowski) [inline components/Addressing/interfaces.rst]: ...api.sylius.org/Sylius/Component/Addressing/**Checker**/RestrictedZoneCheckerInterface...

**Pull request merged**